engine/sql/156_integration.sql

```diff
@@ -0,0 +1,116 @@
+
+UPDATE workflow_run set workflow=replace(workflow::TEXT, '"platform_model_id":', '"integration_model_id":')::jsonb
+WHERE workflow_id in (
+SELECT workflow.id from workflow
+JOIN workflow_node on workflow.id = workflow_node.workflow_id
+JOIN workflow_node_context on workflow_node.id = workflow_node_context.workflow_node_id
+AND workflow_node_context.project_platform_id > 0
+GROUP BY workflow.id)
+AND last_execution>NOW()- INTERVAL '5 DAY';
+
+-- update project_platform and project_platforms attributes
+UPDATE workflow_run set workflow=replace(workflow::TEXT, '"project_platform', '"project_integration')::jsonb
+WHERE workflow_id in (
+SELECT workflow.id from workflow
+JOIN workflow_node on workflow.id = workflow_node.workflow_id
+JOIN workflow_node_context on workflow_node.id = workflow_node_context.workflow_node_id
+AND workflow_node_context.project_platform_id > 0
+GROUP BY workflow.id)
+AND last_execution>NOW()- INTERVAL '5 DAY';
+
+
+-- +migrate Down
+
+ALTER TABLE application_deployment_strategy DROP CONSTRAINT "application_deployment_strategy_pkey";
+ALTER TABLE grpc_plugin DROP COLUMN integration_model_id;
+ALTER TABLE application_deployment_strategy DROP COLUMN project_integration_id;
+ALTER TABLE workflow_node_context DROP COLUMN project_integration_id;
+ALTER TABLE w_node_context DROP COLUMN project_integration_id;
+ALTER TABLE workflow_node_run_job DROP COLUMN integration_plugin_binaries;
+SELECT create_foreign_key('FK_WORKFLOW_NODE_PROJECT_PLATFORM', 'workflow_node_context', 'project_platform', 'project_platform_id', 'id');
+SELECT create_foreign_key_idx_cascade('FK_W_NODE_CONTEXT_PLATFORM', 'w_node_context', 'project_platform', 'project_platform_id', 'id');
+SELECT create_primary_key('application_deployment_strategy', 'application_id,project_platform_id');
+SELECT create_foreign_key_idx_cascade('fk_application_deployment_strategy_platform', 'application_deployment_strategy', 'project_platform', 'project_platform_id', 'id');
+DROP TABLE integration_model;
+DROP TABLE project_integration;
+
+UPDATE grpc_plugin set type = 'platform-deploy_application' WHERE type = 'integration-deploy_application';
+
+UPDATE workflow set workflow_data=replace(workflow_data::TEXT, '"integration_model_id":', '"platform_model_id":')::jsonb;
+UPDATE workflow set workflow_data=replace(workflow_data::TEXT, '"project_integration_id":', '"project_platform_id":')::jsonb;
+
+UPDATE workflow_run set workflow=replace(workflow::TEXT, '"integration_model_id":', '"platform_model_id":')::jsonb
+WHERE workflow_id in (
+SELECT workflow.id from workflow
+JOIN workflow_node on workflow.id = workflow_node.workflow_id
+JOIN workflow_node_context on workflow_node.id = workflow_node_context.workflow_node_id
+AND workflow_node_context.project_platform_id > 0
+GROUP BY workflow.id)
+AND last_execution>NOW()- INTERVAL '5 DAY';
+
+UPDATE workflow_run set workflow=replace(workflow::TEXT, '"project_integration', '"project_platform')::jsonb
```

Comment:
Is it possible to omit some runs in rollback due to interval starting from now, rollback many hours later for example.
Can we just create a new field in migrate up instead of replacing ?

Reply:
seen both